### PR TITLE
Keep Sys.opaque_identity in Cmm and Mach (port upstream PR 9412)

### PR DESCRIPTION
**Pull request by stedolan**
See [upstream](https://github.com/ocaml/ocaml/pull/9412).

### CODE REVIEW

**Review by Gbury — APPROVED** [review]
The flambda2 part is good, but the build failure for i386 (non-exhaustive pattern match in proc.ml) should be fixed before merging.

File: backend/reloadgen.ml [hunk 1]
```diff
@@ -75,6 +75,10 @@ method reload_operation op arg res =
        so that the presence of a probe does not affect
        register allocation of the rest of the code. *)
     (arg, res)
+  | Iopaque ->
+      (* arg = result, can be on stack or register *)
+      assert (arg.(0).stamp = res.(0).stamp);
```
Comment (gretay-js) [inline backend/reloadgen.ml:80]: What forces arg and res to be the same? (I'm sorry about this is a silly question for a PR that has already been reviewed upstream.)

Comment (gretay-js) [reply backend/reloadgen.ml:80]: ok, it's in `selectgen`!

**Review by gretay-js — APPROVED** [review]

**Comment by mshinwell** [comment]
@stedolan Could you please rebase this patch?

**Review by mshinwell — APPROVED** [review]

**Review by xclerc — APPROVED** [review]

**Pull request merged**